File: medical_fe/src/services/roomGroupService.js
import api from "../core/api";

const roomGroupService = {

    // lấy tất cả nhóm phòng
    getAll() {

        return api.get("/room-groups");
    },

    // tạo nhóm phòng
    create(data) {

        return api.post(
            "/room-groups",
            data
        );
    },

    // update nhóm phòng
    update(id, data) {

        return api.put(
            `/ room - groups / ${id} `,
            data
        );
    },

    // xóa nhóm phòng
    delete(id) {

        return api.delete(
            `/ room - groups / ${id} `
        );
    }
};

export default roomGroupService;

File: medical_management_BE/src/main/java/com/nhom2/qnu/service/RoomGroupService.java
package com.nhom2.qnu.service;

import com.nhom2.qnu.model.RoomGroup;

import java.util.List;

public interface RoomGroupService {

    RoomGroup create(RoomGroup request);

    List<RoomGroup> getAll();

    RoomGroup update(String id, RoomGroup request);

    void delete(String id);
}